actors/mailbox.go

```diff
@@ -26,116 +26,190 @@ package actors
 
 import (
 	"sync"
+	"sync/atomic"
 
-	"go.uber.org/atomic"
+	"github.com/tochemey/goakt/pkg/slices"
+	"github.com/tochemey/goakt/pkg/types"
 )
 
+type MailboxHandler func(ReceiveContext)
+
 // Mailbox defines the actor mailbox.
+// It is responsible for storing messages that are to be processed by the actor and is an
+// actor itself. It is also the policy for how messages are enqueued and dequeued from the mailbox.
 // Any implementation should be a thread-safe FIFO
 type Mailbox interface {
```

Comment:
Please don't change the Mailbox interface. It is there to allow anyone to implement a custom mailbox. For instance, you can add your custom implementation using this interface.

Reply:
Also, I would like to know whether you are suggesting changing the mailbox interface.
The interface is there to allow a custom mailbox implementation.